### ISSUE TITLE
Unable to set focus for accessibility VoiceOver

### ISSUE THREAD

**Issue by disha98** [facebook/react-native#30771]
## Description
I'm trying to set the focus order of elements in my screen for voiceover to announce in my Swift+RN project, however there is no provision to do the same. I also wanted to discuss if there's any straightforward way to set the focus to an element when a page loads. 

We have made multiple attempts to use useRef ([Setting focus for accessibility in RN](https://medium.com/dev-genius/setting-focus-for-accessibility-in-react-native-cd7bbe891c5d)) and none of these seem to be working for us.

iOS uses [accessibilityElements](https://developer.apple.com/documentation/objectivec/nsobject/1615147-accessibilityelements) to do the same and allows us to set the order of elements. I'm trying to find something similar on React Native which would help me set the reference.

On the web, screen readers determine the flow of which things are read by looking at the DOM tree and going from top the bottom, and this is the way accessibility is managed in React DOM.

However, on the native side this doesn't seem to be the case.

iOS exposes some APIs to allow developers to control this order, but React Native doesn't seem to expose that to the Javascript side. 

I am displaying a RN module using the bridge from Swift, as a ViewController.

## React Native version:
react-native: 0.63.3 => 0.63.3 
OS: macOS 10.15.7
Xcode: 12.2

## Steps To Reproduce

Navigating to the next screen in my React Native project does not allow me to set the focus to the first element that I want and the focus is incorrect. In cases where a view is presented in top of another view, the focus falls to the background instead. There are also multiple other issues we face in terms of VoiceOver flexibility. 

## Expected Results
Be able to set the focus order explicitly, set focus on a particular element when we navigate to a new page. 

Apart from these issues on iOS, we also face issues on our parallel Android + RN app, where we are missing Keyboard (Physical BT Keyboard) focus (something similar to what is already mentioned in [this issue](https://github.com/facebook/react-native/issues/30464). The issue of setting focus on a new page persists too.

**Comment by nickmcmillan** [comment]
Facing the same problem. The ordering of my layout makes sense visually, but not for folks using Voiceover.

![RPReplay_Final1612138964](https://user-images.githubusercontent.com/3969800/106405033-ed083500-6488-11eb-906b-01defa613997.gif)

Looks like this has been an unresolved issue for a while, here's an issue with a bit more activity: https://github.com/react-native-community/discussions-and-proposals/issues/118

**Comment by amarlette** [comment]
This issue should be resolved by #30888 once that issue is closed as part of the Improved React Native Accessibility Project

**Comment by frags51** [comment]
+1, would appreciate this addition.

**Comment by unyo** [comment]
@nickmcmillan try grouping your elements using a `<ScrollView>` instead of a `<View>` https://github.com/react-native-community/discussions-and-proposals/discussions/389#discussioncomment-7921439
